octoprint_mrbeam/__init__.py

```diff
@@ -2097,7 +2099,23 @@ def set_gcode_deletion(self, enable_deletion):
         self._settings.save()  # This is necessary because without it the value is not saved
         # Everytime the gcode auto deletion is enabled, it will be triggered
         if enable_deletion:
-            self.mrb_file_manager.delete_old_gcode_files()
+            if (
+                self._gcode_deletion_thread is None
+                or not self._gcode_deletion_thread.is_alive()
+            ):
+                self._logger.info(
+                    "set_gcode_deletion: Starting threaded bulk deletion of gcode files."
+                )
+                self._gcode_deletion_thread = threading.Thread(
+                    target=self.mrb_file_manager.delete_old_gcode_files,
+                    name="gcode_deletion_thread",
+                )
+                self._gcode_deletion_thread.daemon = True
+                self._gcode_deletion_thread.start()
```

Comment:
Can be replaced like this:

```python
from octoprint_mrbeam.util import get_thread

self._gcode_deletion_thread = get_thread(daemon=True) (self.mrb_file_manager.delete_old_gcode_files) ()
```

See `util.__init__.py` line 82

Reply:
Also, it the thread necessary now that it's quick?

Reply:
nice. And yes, thread is always a good thing to respond faster to the frontend if possible

Reply:
then the whole function can be decorated with the `@get_thread` decorator.

Reply:
Not exactly. Here I make sure that there is only one file deletion thread running at a time.